### ISSUE TITLE
ParseObject.getRelation returns incomplete ParseRelation

### ISSUE THREAD

**Issue by Mabenan** [parse-community/Parse-SDK-Flutter#603]
When you exectute the following code you get an exception

`
Receipt objectId = Receipt().getObject("objectId");
elementAt.getRelation<Price>("Prices").getQuery();
`
the reason for this is in getRelation the relation is instantiated without setting the attribute _targetClass
![image](https://user-images.githubusercontent.com/4959465/115402328-9db1e500-a1eb-11eb-9f73-1b21227b3540.png)
which is needed in getQuery()
![image](https://user-images.githubusercontent.com/4959465/115402422-b15d4b80-a1eb-11eb-9340-a533cca6b963.png)
The attribute is only set when you execute fromJson or if you  add/remove something from the relation.

Possible Solution would be to add _targetClass in the constructor and/or instance the object when T is set.

**Comment by RodrigoSMarques** [comment]
My sample code:

```dart
    final parseRelation = resultTestAPI.getRelation('productRelation');
    parseRelation.add(ParseObject("Product")..set("objectId", "XGadzYxnac"));

    print('********************************************************');
    print('Save relation - Add Object');
    print('********************************************************');
    parseObject.set<ParseRelation>('productRelation', parseRelation);

    apiResponse = await parseObject.save();

    print(apiResponse.results.toString());

    final relationResponse = await (parseRelation.getQuery()).query();
    if (relationResponse.success) {
      print('sucesso relation ${relationResponse.results.length}');
    }
```

**Comment by Mabenan** [comment]
Yeah but I just want to read, not change the related items.

**Comment by RodrigoSMarques** [comment]
It's in the example, just ignore the save part

```dart
   final parseRelation = resultTestAPI.getRelation('productRelation');
   final relationResponse = await (parseRelation.getQuery()).query();
    if (relationResponse.success) {
      print('sucesso relation ${relationResponse.results.length}');
    }
```

**Comment by Mabenan** [comment]
Yeah but the call getQuery() fails because _targetClass is initial. I currently use whereRelatedTo as a workaround but it would be better if I can use getQuery

**Comment by RodrigoSMarques** [comment]
Strange. I ran my sample code now and an error occurred.
I made the correction in the migration from version 2x to 3x and it was working.
There may have been an error in the merge.
I will evaluate.

**Comment by Tehforsch** [comment]
I have the same problem with `getRelation`. The `ParseRelation` i get back from the `getRelation` call yields a null error when calling `.getQuery()` on it.
I also tried the workaround mentioned by @Mabenan where I used `whereRelatedTo` instead. However, this also fails because the resulting query gives me `results = null` with the following "error":

`Successful request, but no results found`

I tripled check every class name and tried multiple different ways of doing all this but I always end up with either of these two problems. Would be glad if somebody could help out with this.

**Comment by RodrigoSMarques** [comment]
It's really in trouble.  I'm out of time for the last few days to evaluate.  if anyone can verify.

**Comment by Mabenan** [comment]
> I have the same problem with `getRelation`. The `ParseRelation` i get back from the `getRelation` call yields a null error when calling `.getQuery()` on it.
> I also tried the workaround mentioned by @Mabenan where I used `whereRelatedTo` instead. However, this also fails because the resulting query gives me `results = null` with the following "error":
> 
> `Successful request, but no results found`
> 
> I tripled check every class name and tried multiple different ways of doing all this but I always end up with either of these two problems. Would be glad if somebody could help out with this.

Have you tried the resulting query (in the log) in the parse-dashboard. This always helps me to find issues with the query.

**Comment by RodrigoSMarques** [comment]
Fixed in version 3.1.0
